src/erliam_xml.erl

```diff
@@ -12,89 +12,68 @@
-    {Name, case Content of
-               [#xmlElement{} | _] ->
-                   [{ChildName, parse_element(ChildContent)}
-                    || #xmlElement{name = ChildName,
-                                   content = ChildContent} <- Content];
-               _ ->
-                   parse_element(Content)
-           end};
-
+    {Name,
+     case Content of
+       [#xmlElement{} | _] ->
+           [{ChildName, parse_element(ChildContent)}
+            || #xmlElement{name = ChildName, content = ChildContent}
+                   <- Content];
+       _ -> parse_element(Content)
+     end};
 parse_element([#xmlText{} | _] = L) ->
-    case lists:all(fun (E) ->
-                           is_record(E, 'xmlText')
-                   end, L) of
-        true ->
-            unescape_xml_text(xmerl_xs:value_of(L));
-
-        %% if a list of elements is not solely text, discard the text elements and parse
-        %% the non-text elements:
-        false ->
-            parse_element([X
-                           || X <- L,
-                              not is_record(X, 'xmlText')])
+    case lists:all(fun (E) -> is_record(E, xmlText) end, L) of
+      true -> unescape_xml_text(xmerl_xs:value_of(L));
+      %% if a list of elements is not solely text, discard the text elements and parse
+      %% the non-text elements:
+      false -> parse_element([X || X <- L, not is_record(X, xmlText)])
     end;
-
-parse_element(L) when is_list(L) ->
-    [parse_element(X) || X <- L];
-
-parse_element(X) ->
-    X.
-
+parse_element(L) when is_list(L) -> [parse_element(X) || X <- L];
+parse_element(X) -> X.
 
 unescape_xml_text(X) ->
     unescape_xml_text(unicode:characters_to_binary(X, utf8),
-                      [{<<"&lt;">>, <<"<">>},
-                       {<<"&gt;">>, <<">">>},
-                       {<<"&quot;">>, <<"\"">>},
-                       {<<"&apos;">>, <<"'">>},
+                      [{<<"&lt;">>, <<"<">>}, {<<"&gt;">>, <<">">>},
+                       {<<"&quot;">>, <<"\"">>}, {<<"&apos;">>, <<"'">>},
                        {<<"&amp;">>, <<"&">>}]).
 
-unescape_xml_text(X, []) ->
-    X;
-unescape_xml_text(X, [{C, R}| T]) ->
+unescape_xml_text(X, []) -> X;
+unescape_xml_text(X, [{C, R} | T]) ->
     unescape_xml_text(binary:replace(X, C, R, [global]), T).
 
-
 %%%% TESTS
 -ifdef(TEST).
+
 -include_lib("eunit/include/eunit.hrl").
 
 basic_decode_test() ->
-    Data = <<"<GetSessionTokenResponse xmlns=\"https://sts.amazonaws.com/doc/2011-06-15/\">
-  <GetSessionTokenResult>
-   <Credentials>
-    <AccessKeyId>ACCESS_KEY_ID</AccessKeyId>
-    <SecretAccessKey>SECRET_ACCESS_KEY</SecretAccessKey>
-    <SessionToken>SESSION_TOKEN</SessionToken>
-    <Expiration>EXPIRATION</Expiration>
-   </Credentials>
-  </GetSessionTokenResult>
-  <ResponseMetadata>
-   <RequestId>REQUEST_ID</RequestId>
-  </ResponseMetadata>
- </GetSessionTokenResponse>\n">>,
+    Data = <<"<GetSessionTokenResponse xmlns=\"https://sts.amazonaws"
```

Comment:
Really hard to read like this way  =/

Reply:
yeah… I didn't fix it manually because it was just a test, but it's very easy to manually fix this.
I'll do it soon.